### PR TITLE
Lms 1701 extend report to list historical logstore records

### PR DESCRIPTION
**Pull request by stephen-oh**
**Description**
- Extending report to list historical logstore records and updating the filtering to be compatible with both

**Related Issues**
- N/A

**PR Type**
- Enhancement

### CODE REVIEW

File: classes/form/reportfilter_form.php [hunk 1]
```diff
@@ -33,27 +33,44 @@ public function definition() {
...
+            $mform->addElement('select', 'errortype', get_string('errortype', 'logstore_xapi'), $errortypes);
+        }
+
+        $eventnameselect = $mform->addElement('select', 'eventnames', get_string('eventname', 'logstore_xapi'), $eventnames);
+        $eventnameselect->setMultiple(true);
+
+        if ($reportid == XAPI_REPORT_ID_ERROR) {
+            $mform->addElement('select', 'response', get_string('response', 'logstore_xapi'), $responses);
+        } elseif ($reportid == XAPI_REPORT_ID_HISTORIC) {
+            $mform->addElement('text', 'fullname', get_string('user', 'logstore_xapi'));
```
Comment (lzabo) [inline classes/form/reportfilter_form.php]: We can use core's label here.

Comment (stephen-oh) [reply classes/form/reportfilter_form.php]: Well spotted, I'll update this to `userfullname`

File: lang/en/logstore_xapi.php [hunk 2]
```diff
@@ -104,6 +105,8 @@
 $string['norows'] = "No rows to report";
 $string['notificationsnotenabled'] = "Notifications not enabled";
 $string['notificationtriggerlimitnotreached'] = "Notification trigger limit not reached";
+$string['user'] = 'User';
```
Comment (lzabo) [inline lang/en/logstore_xapi.php:108]: remove this one?

**Review by lzabo — APPROVED** [review]

File: lang/en/logstore_xapi.php (hunk 2 above)
Comment (stephen-oh) [reply lang/en/logstore_xapi.php:108]: I was thinking it was best to have a customisable string here as this field will display the username but in the spec it uses the title "user"

File: lang/en/logstore_xapi.php [hunk 3]
```diff
@@ -89,6 +89,7 @@
 $string['info'] = 'Info';
 $string['datetimegmt'] = 'Date/Time (GMT)';
 $string['logstorexapierrorlog'] = 'Logstore xAPI Error Log';
+$string['logstorexapihistoriclog'] = 'Logstore xAPI Historic Log';
```
Comment (gordonmacqueen-lp) [inline lang/en/logstore_xapi.php:92]: I'm not sure this is the correct name for it, but I'm not sure what it should be.
Does anyone else have any other ideas?

Comment (lzabo) [reply lang/en/logstore_xapi.php:92]: Logstore xAPI - Error log's history

Comment (gordonmacqueen-lp) [reply lang/en/logstore_xapi.php:92]: This one is not for errors. It is for Historical logstore events that the user wants to search for to send.

Comment (lzabo) [reply lang/en/logstore_xapi.php:92]: I see, then:
Logstore xAPI log's history

Comment (mlynn-lp) [reply lang/en/logstore_xapi.php:92]: Could it be "Logstore xAPI Historical Events Log" perhaps?

File: classes/form/reportfilter_form.php [hunk 4]
```diff
@@ -33,27 +33,44 @@ public function definition() {
...
+        $reportid = $this->_customdata['reportid'];
         $eventnames = $this->_customdata['eventnames'];
-        $responses = $this->_customdata['responses'];
         $resend = $this->_customdata['resend'];
 
+        if ($reportid == XAPI_REPORT_ID_ERROR) {
+            $errortypes = $this->_customdata['errortypes'];
+            $responses = $this->_customdata['responses'];
+        } elseif ($reportid == XAPI_REPORT_ID_HISTORIC) {
+            $eventcontexts = $this->_customdata['eventcontexts'];
```
Comment (gordonmacqueen-lp) [inline classes/form/reportfilter_form.php:42]: selecting an `eventcontext` in the form doesn't filter the results
image

File: classes/form/reportfilter_form.php (hunk 1 above)
Comment (gordonmacqueen-lp) [inline classes/form/reportfilter_form.php]: entering a username does not filter the results
image

File: classes/form/reportfilter_form.php (hunk 4 above)
Comment (stephen-oh) [reply classes/form/reportfilter_form.php:42]: I have fixed this if you wish to test

File: classes/form/reportfilter_form.php (hunk 1 above)
Comment (stephen-oh) [reply classes/form/reportfilter_form.php]: I have fixed this if you wish to test

Comment (gordonmacqueen-lp) [reply classes/form/reportfilter_form.php]: It's working now 👍

File: classes/form/reportfilter_form.php (hunk 4 above)
Comment (gordonmacqueen-lp) [reply classes/form/reportfilter_form.php:42]: sorted 👍

**Review by gordonmacqueen-lp — APPROVED** [review]
Those items raised last night are working now.

There's a minor issue around lang strings when there are no entries to list but I'll raise a separate ticket for that
image

**Pull request merged**